.github/workflows/update-deps.yml

```diff
@@ -39,6 +43,15 @@ jobs:
           name: k-framework
           authToken: '${{ secrets.CACHIX_PUBLIC_TOKEN }}'
 
+      - uses: haskell/actions/setup@v2
+        id: setup-haskell
+        with:
+          ghc-version: ${{ env.ghc_version }}
```

Comment:
The `cabal freeze` command was previously picking up the newer ghc from the runner installation (which brought a `base` library newer than our requested one). I tried running it under a nix shell but with the same result.
The Haskell installation takes some time and we need to manually change the GHC version when we upgrade but we would anyway want to do that IMHO